yeonjae/programmers-1/기사단원의무기.js

```diff
@@ -0,0 +1,21 @@
+function solution(number, limit, power) {
+  const numbersArr = [];
+
+  for (let current = 1; current <= number; current++) {
+    let divisor = 0;
+
+    for (let i = 1; i <= current / 2; i++) {
+      if (current % i === 0) {
+        divisor += 1;
+      }
+    }
+
+    numbersArr.push(divisor + 1);
+  }
+
+  return numbersArr
+    .map((n) => {
+      return n > limit ? power : n;
+    })
+    .reduce((acc, cur) => acc + cur, 0);
+}
```

Comment:
```suggestion
function solution(number, limit, power) {
  const numbersArr = [];

  for (let i = 1; i <= number; i++) {
    let divisor = 0;

    for (let j = 1; j <= i / 2; j++) {
      if (i % j === 0) {
        divisor += 1;
      }
    }

    numbersArr.push(divisor + 1);
  }

  return numbersArr.reduce(
    (acc, cur) => (cur > limit ? acc + power : acc + cur),
    0
  );
}
```
- 풀이잘봤습니다 🙇‍♂️
- 첫 번째 for문에서 current를 변수로 사용하셨는데 명확한 변수명을 위한 의도는 잘알겠지만, for문에서는 그냥 `i`, `j`를 사용하는 것도 크게 문제가 안될 것 같아요!! 
- 그리고 return문에서 map().reduce()말고 reduce() 하나로 연산 횟수를 줄일 수 있습니다.
- 해당 풀이는 윤우님 풀이와 거의 유사해서 윤우님 코멘트는 확인부탁드립니다. 더 최적화된 풀이를 남겨놨습니다

Reply:
넵 윤우님코멘트 확인해보겠습니다!